<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NdtRansacDlg</class>
 <widget class="QDialog" name="NdtRansacDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>581</width>
    <height>470</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NDT Ransac Plane Dectection</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>420</y>
     <width>341</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget_9">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>320</y>
     <width>511</width>
     <height>27</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_9">
    <item>
     <widget class="QLabel" name="label_9">
      <property name="text">
       <string>Min. Points per component</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QDoubleSpinBox" name="minPtsPerCCSpinBox">
      <property name="toolTip">
       <string>Probability that no better candidate was overlooked during sampling (the lower the better!)</string>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="minimum">
       <double>1.000000000000000</double>
      </property>
      <property name="maximum">
       <double>10000000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>500.000000000000000</double>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>32</x>
     <y>24</y>
     <width>511</width>
     <height>265</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Cell Radius</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="3">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>Minimum points per cell</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="4">
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>NDT feature threshold for plane</string>
      </property>
     </widget>
    </item>
    <item row="2" column="4">
     <widget class="QDoubleSpinBox" name="tuThresholdSpinBox">
      <property name="toolTip">
       <string>Probability that no better candidate was overlooked during sampling (the lower the better!)</string>
      </property>
      <property name="prefix">
       <string>tu=</string>
      </property>
      <property name="decimals">
       <number>6</number>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>0.030000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="4">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Min support cells per primitive</string>
      </property>
     </widget>
    </item>
    <item row="3" column="4">
     <widget class="QSpinBox" name="supportPointsSpinBox">
      <property name="toolTip">
       <string>This is the minimal number of points required for a primitive</string>
      </property>
      <property name="statusTip">
       <string>This is the minimal number of points required for a primitive</string>
      </property>
      <property name="whatsThis">
       <string>This is the minimal number of points required for a primitive</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10000000</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="4">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>cell max distance to primitive</string>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QDoubleSpinBox" name="cellepsilonDoubleSpinBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Maximum distance of samples to the ideal shape</string>
      </property>
      <property name="prefix">
       <string>e = </string>
      </property>
      <property name="decimals">
       <number>3</number>
      </property>
      <property name="maximum">
       <double>1000000000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>0.050000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="3">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>max distance to primitive</string>
      </property>
     </widget>
    </item>
    <item row="5" column="4">
     <widget class="QDoubleSpinBox" name="epsilonDoubleSpinBox">
      <property name="toolTip">
       <string>Maximum distance of samples to the ideal shape</string>
      </property>
      <property name="prefix">
       <string>e = </string>
      </property>
      <property name="decimals">
       <number>5</number>
      </property>
      <property name="maximum">
       <double>1000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>0.001000000000000</double>
      </property>
     </widget>
    </item>
    <item row="6" column="0" colspan="2">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>max normal deviation</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0" colspan="3">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>overlooking probability</string>
      </property>
     </widget>
    </item>
    <item row="7" column="3" colspan="2">
     <widget class="QDoubleSpinBox" name="probaDoubleSpinBox">
      <property name="toolTip">
       <string>Probability that no better candidate was overlooked during sampling (the lower the better!)</string>
      </property>
      <property name="decimals">
       <number>6</number>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>0.001000000000000</double>
      </property>
     </widget>
    </item>
    <item row="1" column="4">
     <widget class="QDoubleSpinBox" name="minPtsPerCellSpinBox">
      <property name="toolTip">
       <string>Probability that no better candidate was overlooked during sampling (the lower the better!)</string>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="minimum">
       <double>3.000000000000000</double>
      </property>
      <property name="maximum">
       <double>10000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>10.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="6" column="3" colspan="2">
     <widget class="QDoubleSpinBox" name="maxNormDevAngleSpinBox">
      <property name="toolTip">
       <string>Maximum deviation from the ideal shape normal vector (in degrees)</string>
      </property>
      <property name="prefix">
       <string>a = </string>
      </property>
      <property name="suffix">
       <string> deg.</string>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>25.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="0" column="4">
     <widget class="QDoubleSpinBox" name="cellRaidiusSpinBox">
      <property name="toolTip">
       <string>Probability that no better candidate was overlooked during sampling (the lower the better!)</string>
      </property>
      <property name="decimals">
       <number>6</number>
      </property>
      <property name="maximum">
       <double>10000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>tuThresholdSpinBox</tabstop>
  <tabstop>supportPointsSpinBox</tabstop>
  <tabstop>cellepsilonDoubleSpinBox</tabstop>
  <tabstop>epsilonDoubleSpinBox</tabstop>
  <tabstop>maxNormDevAngleSpinBox</tabstop>
  <tabstop>probaDoubleSpinBox</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>NdtRansacDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>NdtRansacDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
